src/clr-angular/data/datagrid/providers/selection.spec.ts

```diff
@@ -184,6 +184,27 @@ export default function(): void {
         expect(nbChanges).toBe(3);
       });
 
+      it('does not emit selection change after a filter is applied and selection is cleared', function() {
```

Comment:
This test description doesn't match what the test actually checks. I believe the test is correct and the description is wrong, but I don't understand what we're trying to fix here so I might be off. 😛

Reply:
So, normally you'd expect a 4th change to fire, since the code to preserve selection applies and updates the current. In the case of filters, we skip that. Is there a better way to test this or would it be sufficient to add a more clear test description and comment?